src/panels/CharmsAndActionsPanel/CharmsAndActionsPanel.test.tsx

```diff
@@ -158,13 +160,32 @@ describe("CharmsAndActionsPanel", () => {
             new Error("Error while calling getCharmsURLFromApplications"),
           ),
       );
-    renderComponent(<CharmsAndActionsPanel />, { path, url, state });
+    const {
+      result: { container },
+    } = renderComponent(<CharmsAndActionsPanel />, { path, url, state });
     expect(juju.getCharmsURLFromApplications).toHaveBeenCalledTimes(1);
     await waitFor(() =>
       expect(console.error).toHaveBeenCalledWith(
         CharmsAndActionsPanelLabel.GET_URL_ERROR,
         new Error("Error while calling getCharmsURLFromApplications"),
       ),
     );
+    expect(container.querySelector(".p-panel__title")).toContainHTML(
+      CharmsPanelLabel.PANEL_TITLE,
+    );
+    const getCharmsURLErrorNotification = screen.getByText(
+      new RegExp(
+        CharmsAndActionsPanelLabel.GET_URL_ERROR.replace(
+          /[.*+?^${}()|[\]\\]/g,
+          "\\$&",
```

Comment:
This is needed in order to escape the special characters of regular expressions, in our case "(" and ")". Was wondering if it might make sense to make a utility function out of this and use it instead of `RegExp`?

Reply:
In this particular case I think RTL provides a couple of other ways to check for the substring that means we don't have to escape the error message:

With the `exact` option:
```typescript
    const getCharmsURLErrorNotification = screen.getByText(
      CharmsAndActionsPanelLabel.GET_URL_ERROR,
      { exact: false },
    );
```

With a custom matcher:
```typescript
    const getCharmsURLErrorNotification = screen.getByText((content) =>
      content.startsWith(CharmsAndActionsPanelLabel.GET_URL_ERROR),
    );
```

Reply:
Done. Used the first proposed fix and also used this fix in other places to combat future potential bugs.